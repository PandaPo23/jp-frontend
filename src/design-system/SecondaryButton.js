import { themeGet } from 'styled-system';

import system from '../utils/System';
import Button from './Button';

const SecondaryButton = system(
  'SecondaryButton',
  {
    extend: Button,
    bg: 'secondary',
  },
  (props) => ({
    '&:hover': {
      background: props.disabled
        ? null
        : themeGet(`colors.${props.hoverBg}`)(props),
    },
  })
);

export default SecondaryButton;
