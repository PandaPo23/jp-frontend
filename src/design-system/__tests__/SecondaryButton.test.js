import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import SecondaryButton from '../SecondaryButton';

describe('SecondaryButton component', () => {
  const getComponent = (props) => <SecondaryButton />;
  it('should render as SecondaryButton', () => {
    const component = getComponent();
    const wrapper = mount(component);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
