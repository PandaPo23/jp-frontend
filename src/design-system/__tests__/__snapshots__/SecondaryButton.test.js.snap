// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SecondaryButton component should render as SecondaryButton 1`] = `
.emotion-0 {
  background-color: secondary;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 8px;
  padding-right: 8px;
  display: inline-block;
  min-width: 100px;
  border: 1px;
  border-radius: sm;
  cursor: pointer;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

<SecondaryButton
  as="button"
  bg="secondary"
  border={1}
  borderRadius="sm"
  display="inline-block"
  hoverable={true}
  minWidth={100}
  px={2}
  py={1}
>
  <button
    className="emotion-0"
  />
</SecondaryButton>
`;
